### PR TITLE
Fix makefile compatibility with podman for mac

### PR DESCRIPTION
**Pull request by cblecker**
https://github.com/containers/podman/issues/13631#issuecomment-1077643246

We don't want/need to set the SELinux mount label when running on podman for macOS. Also, when we pull images, make sure we pull them for linux/amd64

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,9 +5,14 @@ SHELL=/usr/bin/env bash -o errexit
 export CONTAINER_ENGINE ?= docker
 export SKIP_PULL ?= false
 
+VOLUME_MOUNT_FLAGS = :z
 ifeq ($(CONTAINER_ENGINE), docker)
 	USER=--user $(shell id -u):$(shell id -g)
 else
+	ifeq ($(shell uname -s),Darwin)
```
Comment (danilo-gemoli) [inline Makefile:12]: There is a missing space between the comma and `Darwin`

**Comment by danilo-gemoli** [comment]
I would create two commits in this PR: one for `--platform linux/amd64` and the other for `:z`

**Comment by cblecker** [comment]
@danilo-gemoli Updated per your suggestions

**Pull request merged**